agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/FirstEntryPoint.java

```diff
@@ -138,6 +140,20 @@ public void init(EarlyInitAgentConfig earlyConfig) {
     }
   }
 
+  private static Map<String, String> findEnvVariables() {
+    Map<String, String> env = System.getenv();
+    return env.entrySet().stream()
+        .filter(entry -> !isSensitiveProperty(entry.getKey()))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  private static boolean isSensitiveProperty(String key) {
+    return key.contains("user")
+        || key.contains("password")
+        || key.contains("pwd")
+        || key.contains("secret");
```

Comment:
maybe worth contains case insensitive?

also, I'd suggest still displaying the env var, but displaying the value as `****`